samples/nrf9160/gps/src/main.c

```diff
@@ -130,85 +133,97 @@ static int activate_lte(bool activate)
 }
 #endif
 
-static int gnss_ctrl(uint32_t ctrl)
+static void gnss_event_handler(int event)
 {
 	int retval;
-
-	nrf_gnss_fix_retry_t    fix_retry    = 0;
-	nrf_gnss_fix_interval_t fix_interval = 1;
-	nrf_gnss_delete_mask_t	delete_mask  = 0;
-	nrf_gnss_nmea_mask_t	nmea_mask    = NRF_GNSS_NMEA_GSV_MASK |
-					       NRF_GNSS_NMEA_GSA_MASK |
-					       NRF_GNSS_NMEA_GLL_MASK |
-					       NRF_GNSS_NMEA_GGA_MASK |
-					       NRF_GNSS_NMEA_RMC_MASK;
-
-	if (ctrl == GNSS_INIT_AND_START) {
-		gnss_fd = nrf_socket(NRF_AF_LOCAL,
-				     NRF_SOCK_DGRAM,
-				     NRF_PROTO_GNSS);
-
-		if (gnss_fd >= 0) {
-			printk("GPS Socket created\n");
+	struct nrf_modem_gnss_nmea_data_frame *nmea_data;
+
+	switch (event) {
+	case NRF_MODEM_GNSS_EVT_PVT:
+		nrf_modem_gnss_read(&last_pvt, sizeof(last_pvt), NRF_MODEM_GNSS_DATA_PVT);
+		new_data = true;
+		break;
+
+	case NRF_MODEM_GNSS_EVT_FIX:
+		got_fix = true;
+		fix_timestamp = k_uptime_get();
+		break;
+
+	case NRF_MODEM_GNSS_EVT_NMEA:
+		nmea_data = k_malloc(sizeof(struct nrf_modem_gnss_nmea_data_frame));
+
+		retval = nrf_modem_gnss_read(nmea_data,
+					     sizeof(struct nrf_modem_gnss_nmea_data_frame),
+					     NRF_MODEM_GNSS_DATA_NMEA);
+		if (retval == 0) {
+			retval = k_msgq_put(&nmea_queue, &nmea_data, K_NO_WAIT);
```

Comment:
It seems this return code is not checked. If that's intentional, it could be `void`ed explicitly

Reply:
That's a bug, `nmea_data` needs to be freed if putting it to the message queue fails, I'll fix it.